strawberryfields/parameters.py

```diff
@@ -161,6 +161,9 @@ def par_evaluate(params):
 
     Args:
         params (Sequence[Any]): parameters to evaluate
+        dtype (np.dtype): NumPy data type to optionally cast parameters
+            to *before* evaluating the symbolic expression. If the parameter
+            is a TensorFlow tensor, the corresponding TensorFlow dtype is used.
```

Comment:
> Only real TF data types require casting to complex, so you're using NumPy dtypes as placeholders for TF dtypes, so the caller does not have to import tensorflow?

Yep. Another option would be to go with strings, but I thought that might be more prone to bugs in the short term? Unless you have another suggestion?

Reply:
A boolean could work here as well, as the only case seems to be real TF data types. If there would be new cases identified (and transitioning to the logic envisioned by solution 2 is still out of reach), then this could be changed to strings perhaps.

Right now the code in the other two files give the impression that `np.complex128` `dtypes` are going to be used, which without prior knowledge of the change in the PR, might lead to confusion.

Reply:
Since this is just a temporary fix I think NumPy types are fine, there's no need to overthink this.